packages/connect/src/callback-client.ts

```diff
@@ -103,8 +103,8 @@ function createUnaryFn<I extends Message<I>, O extends Message<O>>(
       )
       .then(
         (response) => {
-          options?.onHeader?.(response.header);
-          options?.onTrailer?.(response.trailer);
+          options.onHeader?.(response.header);
+          options.onTrailer?.(response.trailer);
```

Comment:
It's nice that the compiler got smarter about narrowing. There are likely more cases that could be simplified, but go unnoticed by the linter.

But I wonder whether this is now a compiler error for users with TS < 5.4 and `skipLibCheck: false`. Can you verify?

Reply:
Verified that it's ok. Here's what I did:
- updated `react/esbuild` in examples-es to use v5.2.2.
- switched out the `createPromiseClient` to use `createCallbackClient`
- built connect-es with this change, ran `npm pack` and moved that to `react/esbuild` node modules
- set `skipLibCheck` to `false` and ran `npm run typecheck`

The above worked successfully, but now I'm curious why. I feel like it _should've_ thrown a compiler error. Is it because we're not exporting the type `UnaryFn`?

Reply:
Now that you say it: lib checking only checks types, and none of the code above ends up in .d.ts files.

Thanks for verifying!